paris-tennis.py

```diff
@@ -0,0 +1,178 @@
+                title='Paris Sport Bot',
+                message="Booked and payed a reservation at {0}, in {1}, {2} days from today".format(hour, name, day),
+                app_icon=None,
+                timeout=10,
+            )
+        else:
+            logging.info(
+                "Held a reservation at {0}, in {1}, {2} days from today - you have 10 minutes to manually pay".format(hour,
+                                                                                                             name,
+                                                                                                             day))
+            notification.notify(
+                title='Paris Sport Bot',
+                message="Held a reservation at {0}, in {1}, {2} days from today - you have 10 minutes to manually pay".format(
+                    hour,
+                    name,
+                    day),
+                app_icon=None,
+                timeout=10,
+            )
+    except Exception as e:
+        logging.error("Failure to book and pay", e)
+
+
+def check_arguments(args):
+    if not (args.Hour and args.Day and args.Name):
+        logging.error("Hour, Day, and Name arguments are mandatory")
+        return False
+
+    if args.Mode:
+        if args.Mode not in BOT_MODES:
+            logging.error("% s is not a valid bot mode" % args.Mode)
+        else:
+            global BotMode
+            BotMode = args.Mode
+
+    if not (int(args.Day) >= 0 and int(args.Day) <= 7):
+        logging.error("% s is not valid day format" % args.Day)
+        return False
+
+    if not (str(args.Hour).endswith("h") and (int(str(args.Hour).split("h")[0]) >= 8
+                                              and int(str(args.Hour).split("h")[0]) <= 20)):
+        logging.error("% s is not valid hour format" % args.Hour)
+        return False
+
+    if not (str(args.Name).replace(" ", "") in PARIS_TENNIS_COURTS):
+        logging.error("% s is not valid paris tennis court name" % args.Name)
+        return False
+
+    dt = datetime.datetime.today() + datetime.timedelta(days=int(args.Day))
+    logging.info("Chosen Mode: % s" % BotMode)
+    logging.info("Chosen Hour: % s" % args.Hour)
+    logging.info("Chosen Day: % s" % dt.day)
+    logging.info("Chosen Court name: % s" % args.Name)
+    return True
+
+
+if __name__ == '__main__':
+    # Read and check arguments from command line
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-m", "--Mode", help="Set bot mode: pay or hold")
+    parser.add_argument("-ho", "--Hour", help="Set the hour to book")
+    parser.add_argument(
+        "-d", "--Day", help="Set in how many days from today to book")
+    parser.add_argument("-n", "--Name", help="Set tennis court name")
+    args = parser.parse_args()
+    if not check_arguments(args):
+        logging.error("Failure to validate arguments")
+        exit(1)
+
+    driver = login(PLAYER_USERNAME, PLAYER_PASSWORD)
```

Comment:
maybe it's a good idea to add exception handling here to catch exceptions coming from login etc..

Reply:
Hum probably I don't understand Python that much, I have to raise the exception inside the method's body?

Reply:
Ah I just saw you already catch the exception and handle it inside the login method. I don't think any further handling is needed.